src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/EmbeddingProvidersConfigImpl.java

```diff
@@ -40,7 +42,13 @@ public ModelConfigImpl(
                 ? Optional.of(grpcModelConfig.getVectorDimension())
                 : Optional.empty(),
             modelParameterList,
-            grpcModelConfig.getPropertiesMap());
+            grpcModelConfig.getPropertiesMap(),
+            grpcModelConfig.hasDeprecated()
+                ? Optional.of(grpcModelConfig.getDeprecated())
+                : Optional.empty(),
+            grpcModelConfig.hasServiceUrl()
+                ? Optional.of(grpcModelConfig.getServiceUrl())
```

Comment:
In this case since `String` is Object, I think you can just use

   Optional.ofNullable(grpcModelConfig.getServiceUrl());

Reply:
- `grpcModelConfig.getServiceUrl()` is assumed to be a non-null value (though I don't see any `@NonNull` annotation or anything like that on that method, it assumes we check if the serviceUrl is present by calling `grpcModelConfig.hasServiceUrl()` before calling the `get` method).
- So, when I change the code to `Optional.ofNullable(grpcModelConfig.getServiceUrl());` the IDE suggests we change this it to `Optional.of(grpcModelConfig.getServiceUrl())` since`grpcModelConfig.getServiceUrl()` is a non-null value.

<img width="979" alt="image" src="https://github.com/user-attachments/assets/36d77794-57d7-4f4a-aa24-3a2af3c1af90">

So, leaving this the way it is